### PR TITLE
Add draft for interactive atomic name swaps

### PR DESCRIPTION
**Pull request by pinheadmz**
This is a bit of a copy/paste from the document merged already into hsd. Everyone was having so much fun with HIPs this week I decided to pile on!

### CODE REVIEW

File: HIP-xxxx.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - This is not enforced by the blockchain, it's just something Alice can do, and should warn Bob about it in step 1
+  - Alice constructs and partially signs a FINALIZE that sends value to her own receive address, and consumes the TRANSFER covenant as an input
+    - This requires `SIGHASH_ANYONECANPAY | SIGHASH_SINGLEREVERSE` so that Bob can still add funds
+  - Alice sends the serialized, partially-signed FINALIZE to Bob as a hex string
+  - Bob decodes the blob, and verifies what he can
+    - His address is in the TRANSFER covenant and the FINALIZE output is valid
+    - Alice's signature is valid, and the value output to her address is the agreed price
+  - Bob adds funds to the TX inputs and adds a change output if necessary
+  - Bob signs his inputs with `SIGHASH_ALL`
+  - Bob broadcasts the final transaction
```
Comment (Falci) [inline HIP-xxxx.md:48]: I'd include here something like "This action transfers the money to Alice and the name to Bob in the same TX. No rollback from this point."

Comment (pinheadmz) [reply HIP-xxxx.md:48]: addressed in ee39317 and rebased on master.

**Comment by Falci** [comment]
Please, use the number `HIP-0004`

**Comment by pinheadmz** [comment]
@Falci 78a20f783b3818bdaff581889ef1a492d9ee112d 🤖: 🚀  🍷

**Comment by pinheadmz** [comment]
rebased on master

**Pull request merged**